axelor-production/src/main/java/com/axelor/apps/production/service/operationorder/OperationOrderOutsourceServiceImpl.java

```diff
@@ -45,18 +69,133 @@ public Optional<Partner> getOutsourcePartner(OperationOrder operationOrder) {
+      ManufOrder manufOrder, PurchaseOrder purchaseOrder, Product product) throws AxelorException {
+
+    BigDecimal quantity = BigDecimal.ONE;
+
+    Unit productUnit = getProductUnit(product);
+
+    // If product unit is appBase.unithours or unitDay or unitMinutes
+    AppBase appBase = appBaseService.getAppBase();
+    if (List.of(appBase.getUnitDays(), appBase.getUnitHours(), appBase.getUnitMinutes())
+        .contains(productUnit)) {
+
+      // Quantity must be computed based on hrDurationPerCycle sums of operationOrders
+      quantity =
+          recomputeQty(
+              manufOrder.getOperationOrderList().stream()
+                  .map(OperationOrder::getPlannedHumanDuration)
+                  .reduce(Long::sum)
+                  .orElse(0L),
+              productUnit);
+    }
+
+    return Optional.ofNullable(
+        purchaseOrderLineService.createPurchaseOrderLine(
+            purchaseOrder, product, null, null, quantity, productUnit));
+  }
 
-    if (operationOrder.getManufOrder().getOutsourcing()
-        || prodProcessLine.getOutsourcing()
-        || operationOrder.getOutsourcing()
-        || (prodProcessLine.getOutsourcable() && operationOrder.getOutsourcing())) {
-      return prodProcessLine.getUseLineInGeneratedPurchaseOrder();
+  @Override
+  public List<PurchaseOrderLine> createPurchaseOrderLines(
+      ManufOrder manufOrder, Set<Product> productSet, PurchaseOrder purchaseOrder)
+      throws AxelorException {
+    Objects.requireNonNull(manufOrder);
+    Objects.requireNonNull(purchaseOrder);
+
+    List<PurchaseOrderLine> list = new ArrayList<>();
+    for (Product product : manufOrder.getProdProcess().getGeneratedPurchaseOrderProductSet()) {
+      PurchaseOrderLine purchaseOrderLine =
+          this.createPurchaseOrderLine(manufOrder, purchaseOrder, product).orElse(null);
+      if (purchaseOrderLine != null) {
+        list.add(purchaseOrderLine);
+        purchaseOrder.addPurchaseOrderLineListItem(purchaseOrderLine);
+      }
+    }
+    return list;
+  }
+
+  protected BigDecimal recomputeQty(Long duration, Unit productUnit) {
+    AppBase appBase = appBaseService.getAppBase();
+
+    // Product is in unit day
+    if (productUnit.equals(appBase.getUnitDays())) {
+      return new BigDecimal(duration)
+          .divide(
+              BigDecimal.valueOf(86400),
+              appBaseService.getNbDecimalDigitForQty(),
+              RoundingMode.HALF_UP);
+    } else if (productUnit.equals(appBase.getUnitHours())) {
+      return new BigDecimal(duration)
+          .divide(
+              BigDecimal.valueOf(3600),
+              appBaseService.getNbDecimalDigitForQty(),
+              RoundingMode.HALF_UP);
+    } else if (productUnit.equals(appBase.getUnitMinutes())) {
+      return new BigDecimal(duration)
+          .divide(
+              BigDecimal.valueOf(60),
+              appBaseService.getNbDecimalDigitForQty(),
+              RoundingMode.HALF_UP);
```

Comment:
small refactor but still interesting I think: put only the value of seconds in a variable, then we can do the BigDecimal conversion and the divide without duplicating it.